apps/charterafrica/src/lib/ecosystem/airtable/processData.js

```diff
@@ -0,0 +1,194 @@
+import * as Sentry from "@sentry/nextjs";
+
+function getter(data, key) {
+  return key ? data?.[key] : null;
+}
+
+function mapSupportersToFields(supporters, config, tableData) {
+  const {
+    schema: { partnerTableColumns },
+  } = config;
+  const { name, url, logo } = partnerTableColumns;
+  const { partners } = tableData;
+  const mapped = supporters.map((id) => {
+    try {
+      const { fields } = partners.find((item) => id === item.id);
+      return {
+        name: getter(fields, name),
+        website: getter(fields, url),
+        logo: getter(fields, logo),
+      };
+    } catch (error) {
+      Sentry.captureMessage(error.message);
+      return null;
+    }
```

Comment:
```suggestion
    const partnerData = partners?.find((item) => id === item.id);
    if (partner) {
      const { fields } = partnerData;
      return {
        name: getter(fields, name),
        website: getter(fields, url),
        logo: getter(fields, logo),
      };
    } else {
      return null;
    }
```

Not sure why we need the try/catch